### ISSUE TITLE
Need to check for array controls in optmize._initialize_krotov_controls

### ISSUE THREAD

**Issue by loganbvh** [qucontrol/krotov#79]
* Krotov version: `1.2.0`
* QuTiP version: `4.5.2`
* Python version: `3.9.1`
* Operating System: Windows 10

### Description
`optimize_pulses` errors out with guess controls specified as an `np.ndarray` because the corresponding key in `pulse_options` is not `control` but rather `id(control)`: https://github.com/qucontrol/krotov/blob/29adff6b461d55c8fc607128d48bcbdd4dd7882d/src/krotov/optimize.py#L650

### What I Did
Run the [transmon state transfer example](https://qucontrol.github.io/krotov/v1.2.0/notebooks/01_example_simple_state_to_state.html) using guess controls in array format:

```
import numpy as np
import qutip
import krotov

tlist = np.linspace(0, 5, 500)

def hamiltonian(omega=1.0, ampl0=0.2):
    """Two-level-system Hamiltonian

    Args:
        omega (float): energy separation of the qubit levels
        ampl0 (float): constant amplitude of the driving field
    """
    H0 = -0.5 * omega * qutip.operators.sigmaz()
    H1 = qutip.operators.sigmax()

    def guess_control(t, args):
        return ampl0 * krotov.shapes.flattop(
            t, t_start=0, t_stop=5, t_rise=0.3, func="blackman"
        )

    guess_array = np.array([guess_control(t, []) for t in tlist])
    return [H0, [H1, guess_array]]

H = hamiltonian()

objectives = [
    krotov.Objective(
        initial_state=qutip.ket("0"), target=qutip.ket("1"), H=H
    )
]

def S(t):
    """Shape function for the field update"""
    return krotov.shapes.flattop(
        t, t_start=0, t_stop=5, t_rise=0.3, t_fall=0.3, func='blackman'
    )

pulse_options = {
    id(H[1][1]): dict(lambda_a=5, update_shape=S)
}

opt_result = krotov.optimize_pulses(
    objectives,
    pulse_options=pulse_options,
    tlist=tlist,
    propagator=krotov.propagators.expm,
    chi_constructor=krotov.functionals.chis_ss,
    info_hook=krotov.info_hooks.print_table(J_T=krotov.functionals.J_T_ss),
    check_convergence=krotov.convergence.Or(
        krotov.convergence.value_below('1e-3', name='J_T'),
        krotov.convergence.check_monotonic_error,
    ),
    store_all_pulses=True,
)
.
.
.
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
C:\anaconda3\envs\sequencing-env\lib\site-packages\krotov\optimize.py in _initialize_krotov_controls(objectives, pulse_options, tlist)
    645     try:
--> 646         guess_controls = [
    647             discretize(

C:\anaconda3\envs\sequencing-env\lib\site-packages\krotov\optimize.py in <listcomp>(.0)
    649                 tlist,
--> 650                 args=(pulse_options[control].get('args', None),),
    651                 via_midpoints=True,

TypeError: unhashable type: 'numpy.ndarray'

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-58-1f905cea4d58> in <module>
     41 }
     42 
---> 43 opt_result = krotov.optimize_pulses(
     44     objectives,
     45     pulse_options=pulse_options,

C:\anaconda3\envs\sequencing-env\lib\site-packages\krotov\optimize.py in optimize_pulses(objectives, pulse_options, tlist, propagator, chi_constructor, mu, sigma, iter_start, iter_stop, check_convergence, info_hook, modify_params_after_iter, storage, parallel_map, store_all_pulses, continue_from, skip_initial_forward_propagation, norm, overlap, limit_thread_pool)
    275         lambda_vals,  # Krotov step width λₐ, for each control
    276         shape_arrays,  # update shape S(t), per control, sampled on intervals
--> 277     ) = _initialize_krotov_controls(objectives, pulse_options, tlist)
    278     if continue_from is not None:
    279         guess_controls, guess_pulses = _restore_from_previous_result(

C:\anaconda3\envs\sequencing-env\lib\site-packages\krotov\optimize.py in _initialize_krotov_controls(objectives, pulse_options, tlist)
    654         ]
    655     except (TypeError, np.ComplexWarning) as exc_info:
--> 656         raise ValueError(
    657             "Cannot discretize controls: %s. Note that "
    658             "all controls must be real-valued. Complex controls must be "

ValueError: Cannot discretize controls: unhashable type: 'numpy.ndarray'. Note that all controls must be real-valued. Complex controls must be split into an independent real and imaginary part in the objectives before passing them to the optimization
```

**Comment by goerz** [comment]
Thank you for reporting this!

We'll work on a fix as soon as we can fix our continuous integration (#82)

**Comment by goerz** [comment]
Should be fixed now. If you're still seeing any issues with the current master branch, please reopen.

We'll put out a bugfix release sometime next week.
